src/drivers/adc/adc.c

```diff
@@ -0,0 +1,88 @@
+#include "FreeRTOS.h"
+#include "drivers/adc/adc.h"
+#include "stm32h7xx_hal.h"
+#include <semphr.h>
+
+extern ADC_HandleTypeDef hadc1;
+
+SemaphoreHandle_t adc_conversion_semaphore;
+SemaphoreHandle_t adc_semaphore;
+
+adc_channel_t current_adc_channel = PROCESSOR_BOARD_VOLTAGE;
```

Comment:
```suggestion
static SemaphoreHandle_t adc_conversion_semaphore;
static SemaphoreHandle_t adc_semaphore;

static adc_channel_t current_adc_channel = PROCESSOR_BOARD_VOLTAGE;
```
as per BARR-C:2018 1.8a